src/extensions/twowaycontactsyncadapter_impl.h

```diff
@@ -214,17 +225,70 @@ const QSet<QContactDetail::DetailType> &defaultIgnorableDetailTypes()
     return types;
 }
 
-void removeIgnorableDetailsFromList(QList<QContactDetail> *dets, const QSet<QContactDetail::DetailType> &ignorableDetailTypes)
+QHash<QContactDetail::DetailType, QSet<int> > getDefaultIgnorableDetailFields()
+{
+    QHash<QContactDetail::DetailType, QSet<int> > rv;
+    rv.insert(QContactDetail::TypePhoneNumber, QSet<int>() << QContactPhoneNumber__FieldNormalizedNumber);
+    return rv;
+}
+
+const QHash<QContactDetail::DetailType, QSet<int> > &defaultIgnorableDetailFields()
+{
+    static QHash<QContactDetail::DetailType, QSet<int> > fields(getDefaultIgnorableDetailFields());
+    return fields;
+}
+
+void removeIgnorableDetailsFromList(QList<QContactDetail> *dets,
+                                    const QSet<QContactDetail::DetailType> &ignorableDetailTypes,
+                                    const QHash<QContactDetail::DetailType, QSet<int> > &ignorableDetailFields)
 {
     // ignore differences in certain detail types
     for (int i = dets->size() - 1; i >= 0; --i) {
         const QContactDetail::DetailType type(dets->at(i).type());
         if (ignorableDetailTypes.contains(type)) {
-            dets->removeAt(i);      // we can ignore this detail
+            dets->removeAt(i); // we can ignore this detail altogether
+        } else {
+            // remove any fields specified as ignorable by the adapter
+            // (eg, if that adapter cannot sync certain fields of some detail types).
+            QContactDetail modD = dets->at(i);
+            if (ignorableDetailFields.contains(type)) {
+                Q_FOREACH (int ignorableField, ignorableDetailFields.value(type)) {
+                    modD.removeValue(ignorableField);
+                }
+            }
+            // remove any fields which should never be synced by any adapter.
+            modD.removeValue(QContactDetail__FieldProvenance);
+            modD.removeValue(QContactDetail__FieldModifiable);
+            if (type == QContactDetail::TypePhoneNumber) {
+                modD.removeValue(QContactPhoneNumber__FieldNormalizedNumber);
```

Comment:
Why have this in 'defaultIgnorableDetailTypes()' list if it's also hardcoded here? (And I agree it should be hardcoded here)